### DISCUSSION TITLE
Pom less tycho product builds using higher profile for IUs

### DISCUSSION THREAD

**Discussion by nadhemB** [eclipse-tycho/tycho#347] [Q&A] [answered]
I m trying to build a structure project consisting of 5 plugins and a product.
In my ide, the run configuration shows that it s using 375 plugins from the installation folder.
What I understood, and I can be completely clueless, is I either need to include a target-platform definition listing all the required bundles and ius  or just add the eclipse release repositories hosting them and I did the latter.
When I try to build the project, I run into this warning:
**Mirror tool: Problems resolving provisioning plan.: [Unable to satisfy dependency from Dressing.123 1.0.0.202110261644 to org.eclipse.equinox.p2.iu; org.eclipse.jetty.plus [10.0.6,10.0.6]**
org.eclipse.jetty is not included in my product manifest file but it s a transitive dependency. the version 10.0.6 requires java 11 . while my BREE is 1.8.
I want to understand, how tycho picks the versions of plugins when its not explicitly mentioned and is there a way to make it pick the versions in rrespect to a certain profile?

after the pomless build fails for the product, I added the pom  in the product folder

	<build>
		<plugins>
            <plugin>
                <groupId>org.eclipse.tycho</groupId>
                <artifactId>tycho-p2-publisher-plugin</artifactId>
                <version>${tycho.version}</version>
                <configuration>
                    <profiles>JavaSE-8</profiles>
                </configuration>
                <executions>
                    <execution>
                        <id>default-publish-osgi-ee</id>
                        <configuration>
                          <profiles>JavaSE-8</profiles>
                        </configuration>
                      </execution>
                </executions>
            </plugin>

            <plugin>
                <groupId>org.eclipse.tycho</groupId>
                <artifactId>tycho-p2-director-plugin</artifactId>
                <version>${tycho.version}</version>
                <executions>
                    <execution>
                        <id>materialize-products</id>
                        <goals>
                            <goal>materialize-products</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>archive-products</id>
                        <goals>
                            <goal>archive-products</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
		</plugins>
	</build>

**Comment by mickaelistria** [comment, marked as answer]
Have you tried playing with https://www.eclipse.org/tycho/sitedocs/target-platform-configuration/target-platform-configuration-mojo.html#executionEnvironment to force an EE in order to resolve dependencies? The EE that's used to resolve dependencies and the bundle BREE are distinct; the BREE is mostly used to infer the complier class level.

**Reply by nadhemB** [reply]
<plugin>
				<groupId>org.eclipse.tycho</groupId>
				<artifactId>target-platform-configuration</artifactId>
				<version>${tycho.version}</version>
				<configuration>
					<executionEnvironment>JavaSE-1.8</executionEnvironment>
					<pomDependencies>consider</pomDependencies>
					<target>
						<artifact>
							<groupId>org.frs.example</groupId>
							<artifactId>target-platform</artifactId>
							<version>1.0.0-SNAPSHOT</version>
						</artifact>
					</target>
					<environments>
						<environment>
							<os>win32</os>
							<ws>win32</ws>
							<arch>x86_64</arch>
						</environment>
					</environments>
				</configuration>
			</plugin>

**Reply by nadhemB** [reply]
this is my target-platform configuration in the parent pom

**Reply by mickaelistria** [reply]
Newer version of Eclipse Platform cannot be resolved against JavaSE-1.8. Some solutions:
* Move your bundles to Java 11 (best)
* Set executionEnvironment to Java 11, in the target-platform-configuration, or -better- in the .target file
* remove executionEnvironment setting and let Tycho pick the current one from your JRE
* Make your target-platform contain a version of Eclipse Platform that's compatible with JavaSE-1.8
